testutils: add skip.UnderBazelWithIssue

stevendanna opened the pull request:
This is to skip individual tests under bazel. This seems a bit more
fine-grained than the broken_in_bazel tag in the bazel configuration
but also allows us to avoid skipping tests that work outside of bazel
in our main test suite.

Release note: None

stevendanna committed:
testutils: add skip.UnderBazel and skip.UnderBazelWithIssue

This is to skip individual tests under bazel. This seems a bit more
fine-grained than the broken_in_bazel tag in the bazel configuration
but also allows us to avoid skipping tests that work outside of bazel
in our main test suite.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62862)
<!-- Reviewable:end -->

stevendanna commented:
motivated by https://github.com/cockroachdb/cockroach/pull/62861

stevendanna committed:
kvserver: use UnderBazelWithIssue for TestPushTxnHeartbeatTimeout skip

This test apepars to work outside of bazel, so let's limit where we
skip it.

Release note: None

rickystewart reviewed (approved):
wfm

stevendanna commented:
bors r=rickystewart
